lib/flash.toit

```diff
@@ -0,0 +1,148 @@
+// Copyright (C) 2022 Toitware ApS. All rights reserved.
+// Use of this source code is governed by an MIT-style license that can be
+// found in the lib/LICENSE file.
+
+import spi
+import reader show Reader
+import writer show Writer
+import gpio
+
+FLASH_5MHZ    ::= 0
+FLASH_10MHZ   ::= 1
+FLASH_20MHZ   ::= 2
+FLASH_26MHZ   ::= 3
+FLASH_40MHZ   ::= 4
+FLASH_80MHZ   ::= 5
+
+class Flash:
+  flash_ := null
+  mount_point/string
+
+  /**
+  Mounts an SD-card as a FAT file system under $mount_point on the $spi_bus without formatting the flash.
+
+  The $cs is the chip select pin for the SD-card holder.
+  */
+  constructor.sdcard
+      --.mount_point/string
+      --spi_bus/spi.Bus
+      --cs/gpio.Pin:
+    if not mount_point.starts_with "/": throw "INVALID_ARGUMENT"
+    flash_ = init_sdcard_ mount_point spi_bus.spi_ cs.num 0 0 0
+
+
+  /**
+  Mounts an SD-card as a FAT file system under $mount_point on the $spi_bus and formats the SD-card
+    with $max_files and $allocation_unit_size if it is not already formatted.
+
+  The $cs is the chip select pin for the SD-card holder.
+  */
+  constructor.sdcard_format
```

Comment:
Is this the most commonly used constructor? Maybe it should be the unformatted ones that get the `_unformatted` suffix on the names?